packages/algorand-js/src/interpreter/txn.ts

```diff
@@ -0,0 +1,307 @@
+import { Address, Transaction } from "algosdk";
+
+import { TealError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { ZERO_ADDRESS } from "../lib/constants";
+import { toBytes } from "../lib/parse-data";
+import { StackElem, TxnField, TxnType } from "../types";
+import { Op } from "./opcode";
+
+// regular bytes
+const bytes = [
+  TxnField.ApprovalProgram,
+  TxnField.ClearStateProgram,
+  TxnField.ConfigAssetUnitName,
+  TxnField.ConfigAssetName,
+  TxnField.ConfigAssetURL,
+  TxnField.ConfigAssetMetadataHash
+];
+
+// 32 byte addresses
+const addr = [
+  TxnField.Receiver,
+  TxnField.CloseRemainderTo,
+  TxnField.VotePK,
+  TxnField.SelectionPK,
+  TxnField.AssetSender,
+  TxnField.AssetReceiver,
+  TxnField.AssetCloseTo,
+  TxnField.ConfigAssetManager,
+  TxnField.ConfigAssetReserve,
+  TxnField.ConfigAssetFreeze,
+  TxnField.ConfigAssetClawback,
+  TxnField.FreezeAssetAccount
+];
+
+// returns default value of txn field
+function defaultSpec (txField: TxnField): StackElem {
+  if (bytes.includes(txField)) {
+    return new Uint8Array(0);
+  }
+  if (addr.includes(txField)) {
+    return ZERO_ADDRESS;
+  }
```

Comment:
checking in a map / set is much faster then iterating over an array. 
I suggest to use `Set` type above, instead of an Array.

Reply:
ok